packages/firestore/test/lite/integration.test.ts

```diff
@@ -1542,6 +1542,106 @@ describe('withConverter() support', () => {
         });
       });
 
+      it('supports optional fields', () => {
+        interface TestObjectOptional {
+          optionalStr?: string;
+          nested?: {
+            requiredStr: string;
+          };
+        }
+
+        const testConverterOptional = {
+          toFirestore(testObj: WithFieldValue<TestObjectOptional>) {
+            return { ...testObj };
+          },
+          fromFirestore(snapshot: QueryDocumentSnapshot): TestObjectOptional {
+            const data = snapshot.data();
+            return {
+              optionalStr: data.optionalStr,
+              nested: data.nested
+            };
+          }
+        };
+
+        return withTestDocAndInitialData(initialData, async docRef => {
+          const testDocRef: DocumentReference<TestObjectOptional> =
+            docRef.withConverter(testConverterOptional);
+
+          await updateDoc(testDocRef, {
+            optionalStr: 'foo'
+          });
+          await updateDoc(testDocRef, {
+            'optionalStr': 'foo'
+          });
+
+          await updateDoc(testDocRef, {
+            nested: {
+              requiredStr: 'foo'
+            }
+          });
+          await updateDoc(testDocRef, {
+            'nested.requiredStr': 'foo'
+          });
+        });
+      });
+
+      it('supports union fields', () => {
+        interface TestObjectUnion {
+          optionalStr?: string;
+          nested?:
+            | {
+                requiredStr: string;
+              }
+            | { requiredStrObject: string };
```

Comment:
May be a better example would be `: {foo: number}`?

Reply:
Done.